packages/opentelemetry-resources/src/platform/node/detectors/EnvDetector.ts

```diff
@@ -47,25 +48,23 @@ class EnvDetector implements Detector {
    * OTEL_RESOURCE_LABELS environment variable. Note this is an async function
    * to conform to the Detector interface.
    */
-  async detect(config: ResourceDetectionConfig = {}): Promise<Resource> {
+  async detect(
+    config: ResourceDetectionConfigWithLogger = { logger: new NoopLogger() }
```

Comment:
you don't need to create here a new NoopLogger, as you already have the logger. This is either the user logger or the one created previously in `detectResources`

Reply:
Fixed. My original thought process was that it might get called from outside of the detectResources function, but this makes more sense.